module-2/backend/week_one.md

```diff
@@ -5,30 +5,58 @@ Fork this respository. Answer the questions to the best of your ability. Try to
 Note: When you're done, submit a PR. 
 
 1. List the five common HTTP verbs and what the purpose is of each verb.
+get, puts, delete, post, patch
 2. What is Sinatra?
+sinatra is the framework we use to create web apps
 4. What is MVC?
+model view controller
 5. Why do we follow conventions when creating our actions/path names in our Sinatra routes?
+allows order to be maintained, and read-ability to outside developers
 6. What types of variables are accessible in our view templates without explicitly passing them?
+instance vars
 7. Given the following block of code, how would I pass an instance variable `count` with a value of `1` to my `index.erb` template?
   
   ```ruby
   get '/horses' do
+    @count = 1
+    @name = "mr. ed"
     erb :index
   end
   ```
 
 8. In the same code block, how would I pass a local variable `name` with a value of `Mr. Ed` to the view?
 9. What's the purpose of ERB?
+allows us to write html with ruby injects
 10. Why do I need a development AND test database?
+test the several phases without risk of destroying your main data
 11. What's responsive design?
+allows sites to be made to respond to a devices screen size
 12. What is CRUD and why is it important?
+create read update delete, it is the common path creating convention for "building" apps
 13. What does HTTP stand for? 
+hyper text transfer protocol
 14. What are the two ways to interpolate Ruby in an ERB view template? What's the difference between these two ways?
+<%= for displaying the result of the code
+<% just runs the code no need for display
 15. What's an ORM?
+object relationship mapping
 16. What's the most commonly used ORM in ruby (Sinatra & Rails)?
+active record
 17. Let's say we have an application with restaurants. There are seven verb + path combinations necessary to provide full CRUD functionality for our restaurant application. List each of the seven combinations, and explain what each is for.
+get '/' gets the root
+get '/restaurant' gets restaurant page
+post '/restaurant' displays that view erb
+get '/restaurant/:id/edit' gets the edit erb
+put '/restaurant/:id' edits the info and saves it, redirecting you to id page
+get '/restaurant/:id gets the individual page /
+delete '/horses/:id DESTROYS!!!!!!!!
 18. What's a migration? 
+where you bring the data base infor to your application
 19. When you create a migration, does it automatically modify your database?
+no
 20. How does a model relate to a database?
+model allows the db to exist for manipulation

```

Comment:
Be careful here. A database can exist without our models, but our models *need* the database, and the relationships in the database to manipulate the data in the database. Our models are classes that correspond to tables in our database. Instances of our models represent specific rows in those tables.